share/picongpu/examples/KelvinHelmholtz/include/picongpu/param/grid.param

```diff
@@ -33,7 +33,34 @@ namespace picongpu
 
         /** equals X
          *  unit: meter */
+
+#ifndef PARAM_FIELDSOLVER
+#   define PARAM_FIELDSOLVER Yee
+#endif
+
 #define DirSplitting 1
+
+#if( PMACC_CUDA_ENABLED != 1 )
+/* DirSplitting is only available for the CUDA accelerator.
+ * Automatically switch back to Yee when we compile for non CUDA accelerator.
+ */
+#   if (PARAM_FIELDSOLVER == 1)
```

Comment:
Why do you check `PARAM_FIELDSOLVER` to be equal to `1`?  Shouldn't that be string equal to `Yee` or `DirSplitting` or etc.?

Reply:
It is not possible to perform string comparism with precompiler. This is a workaround we used already in the past. 
We will work on removing directional splittung from the kHI example. Currently it ia our way to test different components of PIC. 
The mid term goal is to simplify the examples and move such tests to seperate input sets so that the user will not be co fused by such uglyprecompiler hacks.